azure-proto-core-test/SystemAssignedIdentityTests.cs

```diff
@@ -0,0 +1,299 @@
+﻿using azure_proto_core.Resources;
+using NUnit.Framework;
+using System;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+
+namespace azure_proto_core_test
+{
+    public class SystemAssignedIdentityTests : SystemAssignedIdentity
```

Comment:
Generally, having your test class extend the class that is the subject of the test is a pattern to avoid.  For one thing, it prevents your test from extending any test class that might be required for your framework.  For another, it can be confusing which methods are test methods, and which are inherited methods.

Ifif you do need a test class the extends SystemAssignedIdentity, it is generally better to have a separate 'TestSystemAssignedIdentity' class that is used by this class.